adapter/src/main/java/com/twilio/kudu/sql/rel/KuduSortRel.java

```diff
@@ -111,7 +111,7 @@ public void implement(Implementor implementor) {
     }
 
     implementor.groupByLimited = groupBySorted;
-    implementor.sortPkPrefixColumns = sortPkPrefixColumns;
-    implementor.sortPkColumns = sortPkColumns;
+    implementor.sortPkPrefixColumns.addAll(sortPkPrefixColumns);
```

Comment:
How does this help in resolving the error?

Reply:
Since previously we had a reference to the list its possible that the list was shared across queries (I think).